api/types/events/events.go

```diff
@@ -48,8 +48,8 @@ func (m *DatabaseSessionQuery) TrimToMaxSize(maxSize int) AuditEvent {
 	out.DatabaseQuery = ""
 	out.DatabaseQueryParameters = nil
 
-	// Use 10% max size ballast + message size without custom fields.
-	sizeBallast := maxSize/10 + out.Size()
+	// Use 12% max size ballast + message size without custom fields.
+	sizeBallast := maxSize/8 + out.Size()
```

Comment:
Could you explain how changing this from 10% to 12% helps?

Reply:
I added some additional comments to the code but basically MongoDB requires a lot of escaping when converting from binary format that caused the overall size to still exceed the limits even after trimming.

Reply:
Yes, I was more curious how you picked this 12% number and not say, 15% or 20%? Is there a formula or is it more based on the "feel" and to get a bit more buffer? Can we be sure that we won't run into the same issue with size again?

Reply:
Ah, gotcha. This was based on the tests I ran locally. There is an example in the tests. But your question got me thinking about the pathological cases and how I could better handle those. I'm working on an implementation that will be more reliable.